Failure looks genuine; all of these are failing on the same;


- Experimental; https://jenkins.dockerproject.org/job/Docker-PRs-experimental/45326/console
- Janky: https://jenkins.dockerproject.org/job/Docker-PRs/54184/console
- PowerPC: https://jenkins.dockerproject.org/job/Docker-PRs-powerpc/14426/console
- s390x: https://jenkins.dockerproject.org/job/Docker-PRs-s390x/14344/console

```
22:05:06 --- FAIL: TestServiceUpdateNetwork (14.06s)
22:05:06     update_test.go:201: Creating a new daemon
22:05:06     daemon.go:300: [d4dfefb9307a3] waiting for daemon to start
22:05:06     daemon.go:332: [d4dfefb9307a3] daemon started
22:05:06     update_test.go:242: assertion failed: expression is false: len(netInfo.Containers) == 0: Load balancing endpoint still exists in network
22:05:06     daemon.go:284: [d4dfefb9307a3] exiting daemon
```

--- a/daemon/cluster/executor/container/controller.go
+++ b/daemon/cluster/executor/container/controller.go
@@ -369,20 +369,17 @@ func (r *controller) Shutdown(ctx context.Context) error {
 	}
 
 	if err := r.adapter.shutdown(ctx); err != nil {
-		if isUnknownContainer(err) || isStoppedContainer(err) {
-			return nil
+		if !(isUnknownContainer(err) || isStoppedContainer(err)) {
+			return err
 		}
-
-		return err
 	}
 
 	// Try removing networks referenced in this task in case this
 	// task is the last one referencing it
 	if err := r.adapter.removeNetworks(ctx); err != nil {
-		if isUnknownContainer(err) {
-			return nil
+		if !isUnknownContainer(err) {
+			return err
 		}
-		return err
 	}
 
 	return nil

